hail_search/test_search.py

```diff
@@ -16,22 +88,44 @@ async def test_status(self):
             resp_json = await resp.json()
         self.assertDictEqual(resp_json, {'success': True})
 
-    async def test_search(self):
-        search_body = get_hail_search_body(sample_data=FAMILY_2_VARIANT_SAMPLE_DATA)
+    async def _assert_expected_search(self, results, **search_kwargs):
+        search_body = get_hail_search_body(**search_kwargs)
         async with self.client.request('POST', '/search', json=search_body) as resp:
             self.assertEqual(resp.status, 200)
             resp_json = await resp.json()
         self.assertSetEqual(set(resp_json.keys()), {'results', 'total'})
-        self.assertEqual(resp_json['total'], 4)
-        results = [VARIANT1, VARIANT2, VARIANT3, VARIANT4]
+        self.assertEqual(resp_json['total'], len(results))
         self.assertListEqual(
-            [v['variantId'] for v in resp_json['results']], [v['variantId'] for v in results]
+            [v['variantId'] for v in resp_json['results']], [v['variantId'] for v in results],
+        )
+        for i, result in enumerate(resp_json['results']):
+            self.assertDictEqual(result, results[i])
```

Comment:
The ` assertListEqual(resp_json['results], results)` should work.

Reply:
yeah but its impossible to understand the results of the error for a list of dicts - its not clear whats actually different. This is barely less efficient in terms of runtime, but when there is an issue the error is actually helpful